src/cli/task-telemetry.ts

```diff
@@ -0,0 +1,42 @@
+import { getCompilerSystem, getLogger, getStencilCLIConfig } from './state/stencil-cli-config';
+import { checkTelemetry, disableTelemetry, enableTelemetry } from './telemetry/telemetry';
+
+export const taskTelemetry = async () => {
+  const logger = getLogger();
+  const prompt = logger.dim(getCompilerSystem().details.platform === 'windows' ? '>' : '$');
+  const isEnabling = getStencilCLIConfig().flags.args.includes('on');
+  const isDisabling = getStencilCLIConfig().flags.args.includes('off');
```

Comment:
Similar to my comment above, I'm not sure what the appropriate interface to use is here - should the `logger` be pulled from `getStencilCLIConfig`?

Reply:
`getLogger()` does pull from the `getStencilCLIConfig()` singleton - it's a helper method to return that similar to `getCompilerSystem()`. I may also be misreading your question. Are you asking if `logger` should be stored on the singleton, as in philosophically?

Reply:
To me it seems like there's two ways to get a a logger instance:
```
const loggerViaMethod = getLogger();
const loggerViaSingletonAccessor = getStencilCLIConfig().logger;
```
As a user of the singleton, it's not exactly clear to me if there's a preferred way to get the logger without reading the code that the former is a wrapper around the latter.  The ergonomics of pulling data off the singleton seems a little inconsistent to me - for some fields I need to call `getStencilCLIConfig()`, others have a helper function.  

That said, I'm not sure adding a helper function for each private method necessarily helps here. It gives us consistency/parity where there's a helper function for each private field that we expose via an accessor, but it doesn't answer the "which should I use?" question.

That's all a long winded way of saying "I think this is fine as is for now, because I don't have a great solution to offer up ATM". It's something we should be cognizant of as we evolve this class, but no action required ATM

Reply:
Generally you'll see me throughout this PR use the helper functions, until I need to update the value, at which point I'll use: 

`getStencilCLIConfig().logger = updatedLogger`

And the helper methods pretty much only try to help compensate line length. 

Don't worry, I'm not married to this interaction, but at a minimum the singleton definitely helps.